sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -498,3 +499,34 @@ def test_sparse_coder_parallel_mmap():
 
     sc = SparseCoder(init_dict, transform_algorithm='omp', n_jobs=2)
     sc.fit_transform(data)
+
+
+def test_sparse_coder_init_parameters():
+    # Check that parameters are properly initialize and can be set in
+    # accordance with scikit-learn API
+    # Non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/16336
+
+    rng = np.random.RandomState(777)
+    n_features = 3
+    init_dict = rng.randn(10, n_features)
+    data = np.random.rand(50, n_features)
+
+    sc = SparseCoder(dictionary=init_dict)
+    params = sc.get_params()
+
+    assert_allclose(params["dictionary"], init_dict)
+    assert sc.n_components == init_dict.shape[1]
+    assert not hasattr(sc, "components_")
+
+    sc.fit(data)
+    assert_allclose(sc.components_, init_dict)
+
+    next_dict = rng.randn(5, n_features)
+    sc.set_params(dictionary=next_dict)
+    params = sc.get_params()
+
+    assert_allclose(params["dictionary"], next_dict)
+    assert_allclose(sc.components_, init_dict)
+    sc.fit(data)
+    assert_allclose(sc.components_, next_dict)
```

Comment:
It's weird that no `test_common` needs to be updated.

Reply:
So the common tests are skipped because the model required a parameter `dictionary` which is difficult to set because the size will depend on the `X` fed during the different fit in the common test. Thus, we cannot just give a random dictionary at first.